### PR TITLE
Fix up iOS Add to App tests

### PR DESCRIPTION
**Pull request by dnfield**
## Description

A previous engine roll changed the way some of these interfaces worked.  Warnings were issued but not treated as errors.  This adds pedantic warnings and treats warnings as errors, but excludes the GNU warnings.  it updates sites where warnings were being issued, and simplifies the logic for waiting for the semantics notification to happen. It also extends the timeout on that to 30 seconds from 10, which should make this test less flaky.

Also tries to use xcpretty if available so the output is more legible.

## Tests

Updates/fixes tests for iOS Add to App

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: dev/integration_tests/ios_add2app/ios_add2appTests/IntegrationTests.m [hunk 1]
```diff
@@ -112,8 +88,8 @@ - (void)testDualFlutterView {
         (DualFlutterViewController *)navController.visibleViewController;
     GREYAssertNotNil(viewController,
                      @"Expected non-nil DualFlutterViewController.");
-    waitForFlutterSemanticsTree(viewController.topFlutterViewController);
-    waitForFlutterSemanticsTree(viewController.bottomFlutterViewController);
+    [self expectSemanticsNotification:viewController.topFlutterViewController];
+    [self expectSemanticsNotification:viewController.topFlutterViewController];
```
Comment (jmagman) [inline dev/integration_tests/ios_add2app/ios_add2appTests/IntegrationTests.m]: Did you mean bottomFlutterViewController?

File: dev/integration_tests/ios_add2app/build_and_test.sh [hunk 2]
```diff
@@ -10,4 +10,15 @@ popd
 
 pod install
 os_version=$(xcrun --show-sdk-version --sdk iphonesimulator)
-xcodebuild -workspace ios_add2app.xcworkspace -scheme ios_add2appTests -sdk "iphonesimulator$os_version" -destination "OS=$os_version,name=iPhone X" test
\ No newline at end of file
+
+PRETTY="cat"
```
Comment (jmagman) [inline dev/integration_tests/ios_add2app/build_and_test.sh]: What is the xcpretty stuff for?

File: dev/integration_tests/ios_add2app/ios_add2appTests/IntegrationTests.m (hunk 1 above)
Comment (dnfield) [reply dev/integration_tests/ios_add2app/ios_add2appTests/IntegrationTests.m]: Yes - oops.

File: dev/integration_tests/ios_add2app/build_and_test.sh (hunk 2 above)
Comment (dnfield) [reply dev/integration_tests/ios_add2app/build_and_test.sh]: it turns the wall of xcodebuild output into an easier to read format - e.g. the test output now looks like: ```~/src/flutter/dev/integration_tests/ios_add2app/flutterapp ~/src/flutter/dev/integration_tests/ios_add2app
Building com.example.iosAdd2appFlutter for simulator (ios)...
Running Xcode build...                                                  
 ├─Assembling Flutter resources...                           6.6s
 └─Compiling, linking and signing...                         1.0s
Xcode build done.                                            8.9s
Built /Users/dnfield/src/flutter/dev/integration_tests/ios_add2app/flutterapp/build/ios/iphonesimulator/Runner.app.
~/src/flutter/dev/integration_tests/ios_add2app
Analyzing dependencies
Fetching podspec for `Flutter` from `flutterapp/.ios/Flutter/engine`
Fetching podspec for `FlutterPluginRegistrant` from `flutterapp/.ios/Flutter/FlutterPluginRegistrant`
Fetching podspec for `ios_add2app_flutter` from `flutterapp/.ios/Flutter`
DYLD_INSERT_LIBRARIES is already set up for ios_add2appTests, ignored.
Downloading dependencies
Using EarlGrey (1.15.0)
Using Flutter (1.0.0)
Using FlutterPluginRegistrant (0.0.1)
Using ios_add2app_flutter (0.0.1)
Generating Pods project
Integrating client project
Sending stats
Pod installation complete! There are 4 dependencies from the Podfile and 4 total pods installed.
/usr/local/rubygems/xcpretty
▸ Compiling GeneratedPluginRegistrant.m
▸ Building library libFlutterPluginRegistrant.a
▸ Running script '[CP] Check Pods Manifest.lock'
▸ Running script '[CP-User] Run Flutter Build Script'
▸ Linking ios_add2app
▸ Processing Info.plist
▸ Running script '[CP] Embed Pods Frameworks'
▸ Processing Info.plist
▸ Running script '[CP] Check Pods Manifest.lock'
▸ Linking ios_add2appTests
▸ Running script '[CP] Embed Pods Frameworks'
2019-08-14 21:09:29.413 xcodebuild[47089:7345009]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:
/Users/dnfield/Library/Developer/Xcode/DerivedData/ios_add2app-gxshknetbdjmfsgtdboomxpuligc/Logs/Test/Test-ios_add2appTests-2019.08.14_21-09-16--0700.xcresult/1_Test/Diagnostics/ios_add2appTests-C48643CF-C308-405F-A7A8-0BDCFF530327/ios_add2appTests-89296592-4974-4220-8C28-BCA278245BB4/Session-ios_add2appTests-2019-08-14_210929-lRbsV9.log
2019-08-14 21:09:29.413 xcodebuild[47089:7344994] [MT] IDETestOperationsObserverDebug: (68A5732B-A4B0-4144-878E-CCD296CCD73A) Beginning test session ios_add2appTests-68A5732B-A4B0-4144-878E-CCD296CCD73A at 2019-08-14 21:09:29.413 with Xcode 10E1001 on target <DVTiPhoneSimulator: 0x7fc026421270> {
		SimDevice: iPhone X (2D7CEDFA-28AC-4921-91F9-FBCBD22F5D16, iOS 12.2, Shutdown)
} (12.2 (16E226))
2019-08-14 21:09:42.353 xcodebuild[47089:7344994] [MT] IDETestOperationsObserverDebug: (68A5732B-A4B0-4144-878E-CCD296CCD73A) Finished requesting crash reports. Continuing with testing.
All tests
Test Suite ios_add2appTests.xctest started
FlutterTests
    ✓ testDualFlutterView (3.213 seconds)
    ✓ testFullScreenCanPop (2.450 seconds)
    ✓ testHybridView (6.088 seconds)
ViewControllerRelease
    ✓ testReleaseFlutterViewController (0.372 seconds)


	 Executed 4 tests, with 0 failures (0 unexpected) in 12.124 (12.129) seconds

2019-08-14 21:09:57.254 xcodebuild[47089:7344994] [MT] IDETestOperationsObserverDebug: 27.857 elapsed -- Testing started completed.
2019-08-14 21:09:57.254 xcodebuild[47089:7344994] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
2019-08-14 21:09:57.254 xcodebuild[47089:7344994] [MT] IDETestOperationsObserverDebug: 27.857 sec, +27.857 sec -- end
▸ Test Succeeded
```

Comment (dnfield) [reply dev/integration_tests/ios_add2app/build_and_test.sh]: (if you have xcpretty installed)

Comment (jmagman) [reply dev/integration_tests/ios_add2app/build_and_test.sh]: Interesting.  Looks like there's also https://github.com/thii/xcbeautify which might be more up to date.

**Review by jmagman — APPROVED** [review]
LGTM

File: dev/integration_tests/ios_add2app/build_and_test.sh (hunk 2 above)
Comment (dnfield) [reply dev/integration_tests/ios_add2app/build_and_test.sh]: I have no strong preferences there - we're using xcpretty in a few other places already though

**Pull request merged**